File: code/python/pq/book.py
class Book:
    def __init__(self, name, price, author):
        self.name = name
        self.price = price
        self.author = author

    def __str__(self):
        return f"Name: {self.name}, price: {self.price}, author: {self.author}"

    def __lt__(self, other):
        return self.price < other.price

File: code/java/pq/src/main/java/org/swufe/datastructures/Book.java
package org.swufe.datastructures;

import java.util.Objects;

public class Book implements Comparable<Book> {
    private String name;
    private double price;

    private String author;

    public Book(String name, double price, String author) {
        this.name = name;
        this.price = price;
        this.author = author;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public String getAuthor() {
        return author;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    @Override
    public int compareTo(Book o) {
        return Double.compare(price, o.price);
    }


    @Override
    public int hashCode() {
        return Objects.hash(name, price, author);
    }

    @Override
    public String toString() {
        return "Book{" +
                "name='" + name + '\'' +
                ", price=" + price +
                ", author='" + author + '\'' +
                '}';
    }
}
